opa-filter-core/src/main/java/com/bisnode/opa/spring/security/filter/Jwt.java

```diff
@@ -0,0 +1,67 @@
+package com.bisnode.opa.spring.security.filter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.core.AbstractOAuth2Token;
+
+import java.util.Optional;
+
+import static java.util.Objects.isNull;
+import static java.util.function.Predicate.not;
+
+class Jwt {
+    private static final Logger log = LoggerFactory.getLogger(Jwt.class);
+    private final String encoded;
+
+    private Jwt(String encodedToken) {
+        this.encoded = encodedToken;
+    }
+
+    static Optional<Jwt> fromSecurityContext() {
+        return extractAccessToken(SecurityContextHolder.getContext().getAuthentication())
+                .filter(not(String::isBlank))
+                .filter(Jwt::looksLikeJwt)
+                .map(Jwt::new);
+    }
+
+    private static Optional<String> extractAccessToken(final Authentication authentication) {
+        if (isNull(authentication)) {
+            log.warn("Received null authentication");
+            return Optional.empty();
+        }
+
+        if (!(authentication.getCredentials() instanceof AbstractOAuth2Token)) {
+            log.warn("Authentication credentials not an instance of AbstractOAuth2Token.");
+            return Optional.empty();
+        }
+
+        final String accessToken = ((AbstractOAuth2Token) authentication.getCredentials()).getTokenValue();
```

Comment:
Wondering if credentials could be empty, maybe we should check for them too?

Reply:
I haven't encountered such case, but I can add the check, sure.

Reply:
Actually, there's no need. It's checked by 
```java
 if (!(authentication.getCredentials() instanceof AbstractOAuth2Token)) 
```